Fix inconsistent CI workflows.

https://github.com/KitwareMedical/ITKHalideFilters/actions/runs/10426235433/job/28878788922

`build-test-cxx` suceeded on ubuntu-22.04, but failed on macos-12 and the other platforms were canceled. `python-build-workflow` succeeded on macos and windows, but failed on linux so the other platforms were canceled.

Some other notes in the comments of #8.

The issue on linux is a linking error due to Halide and glibc interacting in manylinux. 

https://github.com/KitwareMedical/ITKHalideFilters/actions/runs/10874257586/job/30201463932#step:5:451

Notes at the end like `undefined reference to pow@GLIBC_2.29` or `undefined reference to pthread_create@GLIBC_2.34`.

Halide is built on Ubuntu LTS via Buildbot. It requires glibc 2.34, but the latest manylinux image provides only glibc 2.28. 

https://github.com/pypa/manylinux?tab=readme-ov-file#manylinux_2_28-almalinux-8-based

---

The issue on MacOS is due to Halide GPU only targeting CUDA. Should update CMake to use cuda or metal depending on the target platform.

---

The issue on Windows is unclear; some linkage error: https://github.com/KitwareMedical/ITKHalideFilters/actions/runs/10885362184/job/30202791711#step:6:774

https://open.cdash.org/builds/9907078/configure

I believe this is the root cause of the issues:

> CMake Error: CMake can not determine linker language for target: HalideFilters
> CMake Error in src/CMakeLists.txt:
> Exporting the target "HalideFilters" is not allowed since its linker language cannot be determined

Due to HalideFilters library not having any source files; it only exports generated headers and Halide object file.

--- a/.github/workflows/build-test-package.yml
+++ b/.github/workflows/build-test-package.yml
@@ -18,6 +18,3 @@ jobs:
     uses: InsightSoftwareConsortium/ITKRemoteModuleBuildTestPackageAction/.github/workflows/build-test-package-python.yml@v5.4.0
     secrets:
       pypi_password: ${{ secrets.pypi_password }}
-    with:
-      manylinux-platforms: '["_2_28-x64", "_2_28-aarch64"]'
-      python3-minor-versions: '["9", "10", "11"]'

--- a/.github/workflows/build-test-package.yml
+++ b/.github/workflows/build-test-package.yml
@@ -18,6 +18,3 @@ jobs:
     uses: InsightSoftwareConsortium/ITKRemoteModuleBuildTestPackageAction/.github/workflows/build-test-package-python.yml@v5.4.0
     secrets:
       pypi_password: ${{ secrets.pypi_password }}
-    with:
-      # Halide requires glibc 2.34 which is not available on any current manylinux platform.
-      manylinux-platforms: '[]'

--- a/.github/workflows/build-test-package.yml
+++ b/.github/workflows/build-test-package.yml
@@ -18,3 +18,6 @@ jobs:
     uses: InsightSoftwareConsortium/ITKRemoteModuleBuildTestPackageAction/.github/workflows/build-test-package-python.yml@v5.4.0
     secrets:
       pypi_password: ${{ secrets.pypi_password }}
+    with:
+      manylinux-platforms: '["_2_28-x64", "_2_28-aarch64"]'
+      python3-minor-versions: '["9", "10", "11"]'

--- a/.github/workflows/build-test-package.yml
+++ b/.github/workflows/build-test-package.yml
@@ -18,3 +18,6 @@ jobs:
     uses: InsightSoftwareConsortium/ITKRemoteModuleBuildTestPackageAction/.github/workflows/build-test-package-python.yml@v5.4.0
     secrets:
       pypi_password: ${{ secrets.pypi_password }}
+    with:
+      # Halide requires glibc 2.34 which is not available on any current manylinux platform.
+      manylinux-platforms: '[]'

--- a/.github/workflows/build-test-package.yml
+++ b/.github/workflows/build-test-package.yml
@@ -14,7 +14,7 @@ jobs:
   cxx-build-workflow:
     uses: InsightSoftwareConsortium/ITKRemoteModuleBuildTestPackageAction/.github/workflows/build-test-cxx.yml@v5.4.0
 
-  python-build-workflow:
-    uses: InsightSoftwareConsortium/ITKRemoteModuleBuildTestPackageAction/.github/workflows/build-test-package-python.yml@v5.4.0
-    secrets:
-      pypi_password: ${{ secrets.pypi_password }}
+#  python-build-workflow:
+#    uses: InsightSoftwareConsortium/ITKRemoteModuleBuildTestPackageAction/.github/workflows/build-test-package-python.yml@v5.4.0
+#    secrets:
+#      pypi_password: ${{ secrets.pypi_password }}

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,6 +5,7 @@ set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED YES)
 
 option(Module_HalideFilters_USE_AUTOSCHEDULER "Use auto-schedulers for Halide filters" OFF)
+option(Module_HalideFilters_TEST_GPU "Run GPU tests" OFF)
 
 # Update the following variables to update the version of Halide used
 set(HALIDE_VERSION "18.0.0")

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -45,3 +45,4 @@ set(HalideFilters_SRCS
 itk_module_add_library(HalideFilters ${HalideFilters_SRCS})
 target_include_directories(HalideFilters PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
 target_link_libraries(HalideFilters PUBLIC itkHalideSeparableConvolutionImpl itkHalideGPUSeparableConvolutionImpl)
+set_target_properties(HalideFilters PROPERTIES LINKER_LANGUAGE CXX)

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -19,12 +19,14 @@ itk_add_test(NAME itkHalideDiscreteGaussianImageFilterTest
   9
   )
 
-itk_add_test(NAME itkHalideGPUDiscreteGaussianImageFilterTest
-  COMMAND
-  HalideFiltersTestDriver
-  --compare DATA{CTChest/ReferenceOutput.mha} ${ITK_TEST_OUTPUT_DIR}/Output.mha
-  itkHalideGPUDiscreteGaussianImageFilterTest
-  DATA{CTChest/Input.mha}
-  ${ITK_TEST_OUTPUT_DIR}/Output.mha
-  9
-  )
+if(Module_HalideFilters_TEST_GPU)
+  itk_add_test(NAME itkHalideGPUDiscreteGaussianImageFilterTest
+    COMMAND
+    HalideFiltersTestDriver
+    --compare DATA{CTChest/ReferenceOutput.mha} ${ITK_TEST_OUTPUT_DIR}/Output.mha
+    itkHalideGPUDiscreteGaussianImageFilterTest
+    DATA{CTChest/Input.mha}
+    ${ITK_TEST_OUTPUT_DIR}/Output.mha
+    9
+    )
+endif()

